### PR TITLE
test: rewrite mersenne tests

### PR DESCRIPTION
**Pull request by piotrekn**
Mersenne tests extracted from `random.spec.ts`.

### CODE REVIEW

File: test/mersenne.spec.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  it('should throw an error when attempting to seed() a non-integer', () => {
+    expect(() =>
+      mersenne.seed_array(
+        // @ts-expect-error: non-integer error
+        'abc'
+      )
+    ).toThrowError(
+      Error('seed_array(A) must take array of numbers; is string')
+    );
+  });
```
Comment (ST-DDT) [inline test/mersenne.spec.ts:177]: Not sure, whether we need this test. Isn't this covered by the compiler?

File: test/mersenne.spec.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      });
+
+      for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
+        describe('rand', () => {
+          it('should return a random number without given min / max arguments', () => {
+            const randomNumber = mersenne.rand();
+            expect(typeof randomNumber).toBe('number');
+          });
+
+          it('should return random number from range <min; max)', () => {
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply test/mersenne.spec.ts:148]: Maybe add the test for #479 already (just disabled).

File: test/mersenne.spec.ts (hunk 1 above)
Comment (Shinigami92) [reply test/mersenne.spec.ts:177]: That's a special case, and IMO yes, this should be tested

File: test/mersenne.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shinigami92) [reply test/mersenne.spec.ts:148]: I'm also only familiar with a syntax like this: https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints

[the remark this answers is not in the document]
Comment (piotrekn) [reply test/mersenne.spec.ts:151]: Thank you for pointing this out! I'm familiar with both notations, and the one you guys mention seam to be more common globally 👍🏻 I like your approach, it's actually `min (inclusive) to max (exclusive)`, but I think `[min, max)` will be fine if you don't mind.

Fun fact 📝 using angular bracket is a regional notation, and there is also one notation with reversed square brackets:
https://psychology.fandom.com/wiki/Interval_(mathematics)

[the remark this answers is not in the document]
Comment (piotrekn) [reply test/mersenne.spec.ts:151]: > Maybe add the test for #479 already (just disabled).

I already did 😃
https://github.com/faker-js/faker/blob/2a21868b7c98fc7d9c2d0b2841db4af7594d1318/test/mersenne.spec.ts#L113-L118

File: test/mersenne.spec.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    describe(`random seeded tests ${JSON.stringify(seed)}`, () => {
+      beforeAll(() => {
+        if (Array.isArray(seed)) {
+          mersenne.seed_array(seed);
+        } else {
+          mersenne.seed(seed);
+        }
+      });
+
+      for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
```
[the remark this answers is not in the document]
Comment (piotrekn) [reply test/mersenne.spec.ts:144]: Sorry, my bad. There was no loop in the original code, but I think it is quite important. It is used in `random` and `datatype`, and `datatype` has 25 runs

File: test/mersenne.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ST-DDT) [reply test/mersenne.spec.ts]: > I already did 😃

Only for non seeded. But I was referring to the seeded one.

[the remark this answers is not in the document]
Comment (ST-DDT) [reply test/mersenne.spec.ts]: > I like your approach, it's actually `min (inclusive) to max (exclusive)`, but I think `[min, max)` will be fine if you don't mind.

Yes, you are right. It is exclusive the max value. Can you please change the jsdocs accordingly?

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]
The tests are a little hard to read, but IMO its good enough for now and way better than the previous tests.

Maybe change the `seededRuns` to 

````ts
type SeededRun = {
  seed: number | number[];
  expectations: SeededRunExpectation[];
};
type SeededRunExpectations = {
  args: any[], // not sure about the type here
  expected: number,
  skip?: boolean,
};
````

Then you can use it like this:

````ts
const actual = mersenne.rand(...args);

expect(actual).toEqual(expected);
````

@Shinigami92 What do you think?

Also, IMO we should use `.toBe()` instead of `.toEqual()` as we are dealing with primitive values anyway.

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**